src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops_padded_gemm.cpp

```diff
@@ -1055,30 +1109,40 @@ bool ConvHipImplicitGemmWrwV4R4Xdlops_Padded_Gemm::IsApplicable(const Convolutio
 
     if(!IsIndexRangeLargeEnough(ctx))
         return false;
-    // gemm size
-    {
-        int gemm_g       = -1;
-        int gemm_m       = -1;
-        int gemm_n       = -1;
-        int gemm_k_total = -1;
-        int gemm_extra_m = -1;
-        int gemm_extra_n = -1;
-        int gemm_extra_k = -1;
 
-        std::tie(gemm_g, gemm_m, gemm_n, gemm_k_total, gemm_extra_m, gemm_extra_n, gemm_extra_k) =
-            CalculateGemmSize(ctx, 16, 64, 4);
-
-        if(gemm_extra_m == 0 && gemm_extra_n == 0 && gemm_extra_k == 0)
-            return false;
-        if(!IsValidGridGemmXdlops(gemm_m, gemm_n, gemm_k_total))
-            return false;
-    }
     // this particular EuristicInit is so comprehensive, that if it cannot predict a valid
     // performance config, the problem is probably not applicable
     PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm config;
     config.EuristicInit(ctx);
 
-    return config.IsReallyValid(ctx);
+    if(!config.IsReallyValid(ctx))
+        return false;
+
+    // gemm size
+    int gemm_g           = -1;
+    int gemm_m           = -1;
+    int gemm_n           = -1;
+    int gemm_k_total     = -1;
+    int gemm_m_pad       = -1;
+    int gemm_n_pad       = -1;
+    int gemm_k_total_pad = -1;
+
+    std::tie(gemm_g,
+             gemm_m,
+             gemm_n,
+             gemm_k_total,
+             std::ignore,
+             gemm_m_pad,
+             gemm_n_pad,
+             gemm_k_total_pad,
+             std::ignore) = config.CalculateGemmSizeAndGemmKBlock(ctx);
+
+    // hack: make itself not applicable if padding is not needed, and fall back to other solver
+    // (likely ConvHipImplicitGemmWrwV4R4Xdlops)
+    if(gemm_m_pad == 0 && gemm_n_pad == 0 && gemm_k_total_pad == 0)
+        return false;
+
+    return IsValidGridGemmXdlops(gemm_m, gemm_n, gemm_k_total);
 }
 
```

Comment:
Because you use ```pad``` transformation instead of really make the matrix C larger in memory, it doesn't require extra memory than the original weight tensor